SNT/NativeMemoryManager.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System.Buffers;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System;
+
+namespace Torch.SNT
+{
+    /// <summary>
+    ///   Class used to wrap the native memory of a Torch Tensor.  
+    /// </summary>
+    internal class NativeMemory<T> : MemoryManager<T>
+    {
+        private int refCount = 0;
+        private IntPtr memory;
+        private int length;
+
+        public NativeMemory(IntPtr memory, int length)
+        {
+            this.memory = memory;
+            this.length = length;
+        }
+
+        public unsafe NativeMemory(void* memory, int length)
+        {
+            this.memory = (IntPtr)memory;
```

Comment:
Can this conversion fail? Also, this constructor could call the other one.

Reply:
I redirected the constructor to the previous one. The conversion is ok, I checked online and it seems that it cannot fail, it is simply boxing a pointer into a managed object.